Fixes Maxmind database download breakage

davidlapsleyio opened the pull request:
Adds a --build-arg variable MAXMIND_LICENSE_KEY which is then used
to automatically download Maxmin dbs.

Requires user to set MAXMIND_LICENSE_KEY variable in their build
environment (needs to be an active license key associated with the
users Maxmind account -- these are available for free).

This commit fixes #9.

corelit committed:
Fixes Maxmind database download breakage

Adds a --build-arg variable MAXMIND_LICENSE_KEY which is then used
to automatically download Maxmin dbs.

Requires user to set MAXMIND_LICENSE_KEY variable in their build
environment (needs to be an active license key associated with the
users Maxmind account -- these are available for free).

This commit fixes #9.

corelit committed:
Fixes issue with multi-stage build and args

corelit committed:
Add a README

corelit committed:
Clean up README.

JustinAzoff commented on 3.0.0.Dockerfile:
```diff
@@ -23,6 +23,7 @@ RUN ${WD}/common/buildbro zeek ${VER}
+ARG MAXMIND_LICENSE_KEY
 RUN apt-get update && apt-get -y install wget ca-certificates --no-install-recommends
 ADD ./common/getmmdb.sh /usr/local/bin/getmmdb.sh
 RUN /usr/local/bin/getmmdb.sh
```
I think should be /usr/local/bin/getmmdb.sh ${MAXMIND_LICENSE_KEY} to pass the key as the first argument.  or need

    ENV MAXMIND_LICENSE_KEY=${MAXMIND_LICENSE_KEY}

otherwise the arg doesnt automatically become an env var

davidlapsleyio replied:
I tested the current commit and it worked even though the ARG was not propagated via an ENV variable.

Having said that, I do prefer the model of passing the license key via command line and then testing it, so I've implemented that in the next commit :)

JustinAzoff commented on README.md:
```diff
@@ -0,0 +1,94 @@
+# Zeek Docker
```
yay!

davidlapsleyio replied:
👍

JustinAzoff commented on common/getmmdb.sh:
```diff
@@ -1,13 +1,61 @@
+}
+
+
+main () {
```
this should check if the license key is set before trying to download files, and if not set, just do nothing.

davidlapsleyio replied:
👍  I've added the check in the next commit.

corelit committed:
Optimizations

- Pass license key in via the getmmdb.sh command line.
- Check the value of license key in the script and do nothing if it is
  blank.
- Add workarounds to 3.0.0.Dockerfile to account for the final build
  stage copy command failing if /usr/share/GeoIP is empty.

corelit committed:
Fix typo.
